core/http/src/main/java/org/trellisldp/http/impl/PostHandler.java

```diff
@@ -152,8 +153,10 @@ public ResponseBuilder createResource() {
                 // Check for any constraints
                 checkConstraint(dataset, PreferUserManaged, ldpType, rdfSyntax.orElse(TURTLE));
             }
-
-            if (resourceService.create(internalId, session, ldpType, dataset.asDataset()).get()) {
+            final IRI container = resourceService.getContainer(internalId).orElse(null);
```

Comment:
This is perfectly fine as it is, but it might be slightly more efficient (and clear, code-wise) to create the container from data we've already got in the `POST` operation. I.e.:

```java
final IRI container = rdf.createIRI(TRELLIS_DATA_PREFIX + req.getPath());
```